docs/site/content/docs/latest/designs/package-process.md

```diff
@@ -454,7 +454,133 @@ spec:
 * `spec.template.spec.fetch[0].imgpkgBundle.image`: THe URL of the location of this package in an OCI registry.
   This value is obtained from the result of the `imgpkg push` command.
 
-### 8. Package Metadata
+### 8. Generate openAPIv3 schema and embed it in a package
+
+Follow the below mentioned steps to get started on generating openAPIv3 schema and specifying it in a package:
+
+1. Create a schema file for given data values file. In ytt, before a Data Value can be used in a template, it must be declared. This is typically done via Data Values Schema
+   Check out [How to write Schema](https://carvel.dev/ytt/docs/latest/how-to-write-schema/), to explore the different annotations that can be used when writing a schema.
+
+2. Run this command to use a data value and schema file together. You can see if you change a value in the `values.yml` to be the incorrect type, you will see an error catches this.
+
+   Use the following template.yaml file
+
+    ```yaml
+    #@ load("@ytt:data", "data")
+    ---
+    data_values: #@ data.values
+    ```
+
+   Let's use secretgen-controller package as an example to generate schema and embed it in Package
+
+   ```bash
+   cd ~/community-edition/addons/packages/secretgen-controller/0.7.1/bundle/config
+   ```
+
+   You can now generate and verify data values by using the command below
+
+   `ytt -f values.yaml -f schema.yaml -f template.yaml`
+
+3. Generate the OpenAPI schema from the `schema.yaml` file.
+
+   `ytt -f schema.yaml --data-values-schema-inspect -o openapi-v3 > openapi-schema.yaml`
+
+4. Create package-template.yaml file from package.yaml by adding the following 2 lines in `spec` field:
```

Comment:
We should have done this when we create the schema of each package and put the `package-template.yaml` file inside the bundle. The developers don't need to worry about this step later on.

Reply:
Can we update this README as we don't need to create template anymore

Reply:
Yes , I am updating it now and also fixing lint issues @blc1996 
I had put a note above that I will change it once initial review for make targets is done.

Reply:
Done